File: python/raindrops/raindrops.py
# Based on my solution at the JavaScript track: https://exercism.org/tracks/javascript/exercises/raindrops/solutions/thuyencode

RAINDROPS = tuple({3: "Pling", 5: "Plang", 7: "Plong"}.items())


def convert(number: int) -> str:
    """Convert `number` into its corresponding raindrop sounds."""
    if number <= 1:
        return str(number)
    raindrop_sounds = ""

    for raindrop in RAINDROPS:
        if number % raindrop[0] == 0:
            raindrop_sounds += raindrop[1]

    return raindrop_sounds if len(raindrop_sounds) != 0 else str(number)

File: solutions/javascript/raindrops/1/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

/**
 * Convert a number into a string that contains raindrop sounds corresponding to certain potential factors.
 *
 * @param {number} num
 */
export const convert = (num) => {
  if (num <= 1) return String(num)

  const raindrops = Object.keys(RAINDROPS).reduce(
    (acc, factor) =>
      num % Number(factor) === 0 ? acc + RAINDROPS[factor] : acc,
    ''
  )

  return raindrops.length !== 0 ? raindrops : String(num)
}

const RAINDROPS = {
  3: 'Pling',
  5: 'Plang',
  7: 'Plong'
}
